hs_core/hydroshare/resource.py

```diff
@@ -369,31 +369,30 @@ def create_resource(
 
         add_resource_files(resource.short_id, *files)
 
-        if 'owner' in kwargs:
-            owner = utils.user_from_id(kwargs['owner'])
-            resource.owners.add(owner)
-
-        resource.owners.add(owner)
+        resource_access = ResourceAccess(resource=resource)
+        resource_access.save()
+        UserResourcePrivilege(resource=resource_access, grantor=owner.uaccess, user=owner.uaccess,
+                              privilege=PrivilegeCodes.OWNER).save()
 
         if edit_users:
             for user in edit_users:
                 user = utils.user_from_id(user)
-                resource.edit_users.add(user)
```

Comment:
I am OK taking the time to get this integrated properly in one step.  I think that this is better than running two access control mechanisms side by side in an operational deployment.  Now side by side may be needed while we are developing, but lets see if we can get this clean for the next operational deploy, and if that has to wait a bit, then that is OK (as long as we know what the wait is)